lib/graphql/schema/argument.rb

```diff
@@ -186,6 +188,15 @@ def prepare_value(obj, value, context: nil)
           raise "Invalid prepare for #{@owner.name}.name: #{@prepare.inspect}"
         end
       end
+
+      private
+
+      def check_conflicting_params
+        return unless !@null && default_value?
+
+        raise ArgumentError, "Argument '#{@name}' has conflicting params! " \
+          "Arguments can't be required and have default values at the same time!"
```

Comment:
The one thing I'd add is a clue to the developer _how_ to fix the error, for example: 

```suggestion
          "Arguments can't be required and have default values at the same time, either use `required: false` or remove `default_value:`."
```